src/ui/components/common/nav.rs

```diff
@@ -1,61 +1,155 @@
-use leptos::{component, view, IntoView};
+use crate::i18n::*;
+use leptos::{component, create_server_action, server, view, IntoView, ServerFnError};
 use leptos_icons::*;
 use leptos_router::*;
 
+#[server(LogoutFormFn, "/api")]
+pub async fn logout_form_fn() -> Result<(), ServerFnError> {
+  use crate::api::set_cookie_wrapper;
+  use leptos_actix::redirect;
+
+  redirect("/");
+
+  match set_cookie_wrapper("jwt", "").await {
+    Ok(o) => Ok(o),
+    Err(e) => Err(ServerFnError::ServerError(e.to_string())),
+  }
```

Comment:
I would assume the `remove_cookie_wrapper` that was made in the other file is for this purpose. This all may be moot if we go with actix-session for cookie management.